### DISCUSSION TITLE
Quickfix or similar in core?

### DISCUSSION THREAD

**Discussion by askreet** [helix-editor/helix#5902] [Ideas]
Has there been previous discussion (sorry if I couldn't find it on GitHub) about adding something like quickfix to core?

Specific things I'd like to see, personally:
1. All pickers supporting a "go to quickfix" hotkey (e.g. C-q) that saves their search results in a format that I can jump between with `]q` + `[q`.
2. A command that can run other commands and populate a quickfix list from known formats (I believe the analogous vim command is `cexec`).
3. Commands to toggle display of the quickfix list.

Edit: I've got a prototype started [here](https://github.com/askreet/helix/tree/quickfix). Currently supports taking any "FilePicker" and pressing C-q to slurp the filtered items into the quickfix list, navigable by ]q and [q. Works with both <kbd>&lt;space&gt;/</kbd> and <kbd>gr</kbd>, for example. I plan to keep using it personally for a bit since it was critical to my vim workflow but super interested in any feedback if others try running it.

**Comment by lxcode** [comment]
The biggest use case for me is vim's :grep command, which I use heavily. While a ripgrep "picker" might or might not make sense, a simple call out to rg that populates the qf would be great.

**Reply by robinmoussu** [reply]
For me it’s `:make`, and I would love to see any progress on this proposal.

**Comment by lxcode** [comment]
Just tried out your branch and it works pretty smoothly. The ability to populate from pickers is pretty flexible —  I particularly like that you can use ```<space>d``` to populate the list. Also upon testing, I think that ```<space>/``` actually obviates the need for a ```:grep``` command.

I was going to suggest having ```]q``` / ```[q``` jump to the matching text/symbol instead of EOL (and beginning of line if there's no actual symbol or substring to match), but having the full line selected is also useful in its own right, and may be more Helix-y.

Nice work, and would love to see this get fleshed out with an actual quickfix window and put into core.

**Comment by theunixer** [comment]
Hello, is  there any progress? A lot of people need this feature.

**Comment by bradjabel** [comment]
The ability to add an entire search list to a quickfix list, or individual items from a search list, would be insane. Really enjoying this editor and watching it improve.

**Comment by tgross35** [comment]
Is this the same as LSP-provided fixes like what "ctrl+." does in VSCode? I am not sure how the file picker mentioned in the top post comes into play.

**Comment by thomasgoulet** [comment]
Maybe I can bump this one to resuscitate it :)

Would it be possible to simply have a command to open the next entry (and previous entry) from the last picker ? Most people already use the `:last_picker <tab><enter>` to go through the list one by one. This could be really useful to include in macros to go through your file picker items automatically.

**Reply by markstos** [reply]
I'm submitted a PR to at least improve the docs to mention that last_picker restores the selected entry. I had never tried last_picker until I read this comment because from reading the current docs I assumed it did exactly what opening the last picker again does!

https://github.com/helix-editor/helix/pull/10843

**Comment by peter-lejon-v** [comment]
While we are on the topic, It would be really amazing if it could be possible to add a feature like [wgrep](https://github.com/mhayashi1120/Emacs-wgrep) to the quickfix list as well! I think it fits nicely with the kakoune philosophy :)

**Reply by jla2000** [reply]
So basically like :cdo in neovim? :)

**Reply by zetashift** [reply]
Related PR: https://github.com/helix-editor/helix/pull/4381

**Reply by trev-dev** [reply]
> So basically like :cdo in neovim? :)

You mean Vim ;)

**Comment by CelestialCrafter** [comment]
any updates on this? it doesn't seem like anything quickfix-related is going anywhere.

**Comment by LunarLambda** [comment]
I would also love this, coming from 8 years of vim.
